src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs

```diff
@@ -0,0 +1,171 @@
+            var ifIndex = parentBlock.Statements.IndexOf(ifStatement);
+            if (ifIndex == 0)
+            {
+                return;
+            }
+
+            var previousStatement = parentBlock.Statements[ifIndex - 1];
+            if (!previousStatement.IsKind(SyntaxKind.LocalDeclarationStatement))
+            {
+                return;
+            }
+
+            var localDeclarationStatement = (LocalDeclarationStatementSyntax)previousStatement;
+            var variableDeclaration = localDeclarationStatement.Declaration;
+
+            if (variableDeclaration.Variables.Count != 1)
+            {
+                return;
+            }
+
+            var declarator = variableDeclaration.Variables[0];
+            if (declarator.Initializer == null)
+            {
+                return;
+            }
+
+            if (!Equals(declarator.Identifier.ValueText, conditionName.Identifier.ValueText))
+            {
+                return;
+            }
+
+            // Syntactically this looks good.  Now make sure that the local is a delegate type.
+            var semanticModel = syntaxContext.SemanticModel;
+            var localSymbol = (ILocalSymbol)semanticModel.GetDeclaredSymbol(declarator);
+            if (localSymbol.Type.TypeKind != TypeKind.Delegate)
+            {
+                return;
+            }
+
+            // Ok, we made a local just to check it for null and invoke it.  Looks like something
+            // we can suggest an improvement for!
+            // But first make sure we're only using the local only within the body of this if statement.
+            var analysis = semanticModel.AnalyzeDataFlow(localDeclarationStatement, ifStatement);
+            if (analysis.ReadOutside.Contains(localSymbol) || analysis.WrittenOutside.Contains(localSymbol))
+            {
+                return;
+            }
+
+            // Looks good!
+            var tree = semanticModel.SyntaxTree;
+            var additionalLocations = new List<Location>
+            {
+                Location.Create(tree, localDeclarationStatement.Span),
+                Location.Create(tree, ifStatement.Span),
+                Location.Create(tree, expressionStatement.Span)
+            };
+
+            syntaxContext.ReportDiagnostic(Diagnostic.Create(descriptor,
+                Location.Create(tree, TextSpan.FromBounds(localDeclarationStatement.SpanStart, invocationExpression.SpanStart)),
+                additionalLocations));
+
+            if (localDeclarationStatement.Span.End != ifStatement.Span.End)
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(descriptor,
+                    Location.Create(tree, TextSpan.FromBounds(expressionStatement.Span.End, ifStatement.Span.End)),
+                    additionalLocations));
+            }
+        }
+
+        private bool IsNotEqualsTest(ExpressionSyntax left, ExpressionSyntax right) =>
```

Comment:
But it's the test if this is the "not equals" expression.  How about "IsNotEqualsBinaryExpression?